packages/node_modules/@webex/webex-core/src/lib/services/service-host.js

```diff
@@ -0,0 +1,250 @@
+   * Get the cluster value.
+   *
+   * @returns {string} - The cluster value.
+   */
+  get cluster() {
+    return this.id.split(':')[2];
+  }
+
+  /**
+   * Get the group value.
+   *
+   * @returns {string} - The group value.
+   */
+  get group() {
+    return this.id.split(':')[1];
+  }
+
+  /**
+   * Get the head value.
+   *
+   * @returns {string} - The head value.
+   */
+  get head() {
+    return this.id.split(':')[0];
+  }
+
+  /**
+   * Determine if the host is local to the user's cluster.
+   *
+   * @returns {boolean} - If the host is local.
+   */
+  get local() {
+    return this.default.includes(this.hostGroup);
+  }
+
+  /**
+   * Get the service value.
+   *
+   * @returns {string} - The service value.
+   */
+  get service() {
+    return this.id.split(':')[3];
+  }
+
+  /**
+   * Get the formatted url for the host.
+   *
+   * @returns {string} - The service url.
+   */
+  get url() {
+    // Generate a url object from the default url.
+    const urlObj = Url.parse(this.default);
+
+    // Format the host of the generated url object.
+    urlObj.host = `${this.uri}${urlObj.port ? `:${urlObj.port}` : ''}`;
+
+    // Assign the formatted url to this.
+    return Url.format(urlObj);
+  }
+
+  /**
+   * The parameter transfer object for {@link setStatus}.
+   *
+   * @typedef {Object} SetStatusPTO
+   * @property {boolean} SetStatusPTO.failed - The failed status to set.
+   * @property {boolean} SetStatusPTO.replaced - The replaced status to set.
+   */
+
+  /**
+   * Set one of the status properties of the class object.
+   *
+   * @param {SetStatusPTO} pto - The parameter transfer object.
+   * @returns {this}
+   */
+  setStatus({failed, replaced}) {
+    if (failed || failed === false) {
+      this.failed = failed;
+    }
+
+    if (replaced || replaced === false) {
+      this.replaced = replaced;
+    }
+
+    return this;
+  }
+
+  /**
+   * The parameter transfer object for {@link polyGenerate}.
+   *
+   * @typedef {object} PolyGeneratePTO
+   * @property {object} PolyGeneratePTO.catalog - The target catalog.
+   * @property {string} PolyGeneratePTO.name - The service name.
+   * @property {string} PolyGeneratePTO.url - The service url.
+   */
+
+  /**
+   * Generate a service host using only a catalog, name, and URL.
+   *
+   * @param {PolyGeneratePTO} pto - The parameter transfer object.
+   * @returns {ServiceHost} - The generated service host.
+   */
+  static polyGenerate({catalog, name, url}) {
+    return new ServiceHost({
+      catalog,
+      defaultUri: url,
+      hostGroup: Url.parse(url).host,
+      id: (name) ? `poly-head:poly-group:poly-cluster:${name}` : undefined,
+      priority: 1,
+      uri: Url.parse(url).host
+    });
+  }
+
+  /**
+   * Validate that a constructor parameter transfer object is valid.
+   *
+   * @param {ConstructorPTO} pto - The parameter transfer object.
+   * @throws - If the parameter transfer object is not valid.
+   * @returns {void}
+   */
+  static validate(pto) {
```

Comment:
Simple validation method to make sure the constructor has all of the information that it needs.